js/MVP/views/landing/Landing.js

```diff
@@ -59,6 +60,18 @@ const Landing = () => {
     );
   };
 
+  const harIngenData = () => {
```

Comment:
Må man ha så mange && sjekker for å sjekke om man ikke har sykmelding? Er det mulig å ha 1 sjekk på dette, i stedet for å både sjekke motebehov, brev, møteplanlegger samtidig?

Ellers kan det være greit å skrive unit tests på denne dersom den må ha så mye logikk

Reply:
Ref samtalen vi hadde tidligere i dag:

Her er noen punkter:

- Vi skal kunne vise `<FetchFailedError />`   samtidig med annen content. Det var besluttet tidligere at hvis noen endepunkter er nede, viser vi det vi kan. Feks hvis behov og planlegger feiler, kan vi vise referat.

`displayContent()` kan være sånn da:
```
  const displayContent = () => {
    if (harIngenData()) {
      return <IkkeSykmeldtLanding />;
    }
    return (
      <>
        {displayMotebehov() && <MotebehovPanel motebehov={motebehov} />}

        <DialogmoteFeaturePanel />
        <PreviousMotereferatFeaturePanel />
      </>
    );
  };
```

- Ang å forenkle` harIngenData()` . Jeg tenker mest på scenario når feks planlegger og motebehov feiler (eller returnerer ingen data) mens brev feiler men potensielt skulle returnere noe, skal det vises både feilinfoboks og "ikke sykmeld" tekst hvis vi bruker ditt forslag. Det kan være forvirrende for bruker som kanskje allerede vet at h#n er sykmeldt å se denne teksten, Siden det å sjekke !brev.data   kan bety enten at endepunkt feiler eller at endepunkt returnerte [] synes jeg at det bør være status sjekk her slik at vi trolig sjekker om det er ingen data pga manglende sykmelding og ikke pga server error.

NB: moteplanlegger returnerer 404 når det er ingen møte, så det er egentlig ikke feil i de fleste tilfeller.

Mitt forslag er:

  ```
const harIngenData = () => {
    return (
      motebehov.isSuccess &&
      !motebehov.data.visMotebehov &&
      brev.isSuccess &&
      !brev.data &&
      ((moteplanlegger.isError && moteplanlegger.error.message === '404') ||
        (moteplanlegger.isSuccess && !moteplanlegger.data))
    );
  };
```